server/rosetta/cosmos/conversion/from_rosetta.go

```diff
@@ -57,41 +60,47 @@ func RosettaAmountsToCoins(amounts []*types.Amount) sdk.Coins {
 	return feeCoins
 }
 
-func ConvertOpsToMsgs(ops []*types.Operation) ([]sdk.Msg, string, error) {
+func ConvertOpsToMsgs(ir types2.InterfaceRegistry, ops []*types.Operation) ([]sdk.Msg, string, error) {
 	var msgs []sdk.Msg
 	var signAddr string
-	var sendOps []*types.Operation
-	var delOps []*types.Operation
+	var operationsByType = make(map[string][]*types.Operation)
 	for _, op := range ops {
-		switch op.Type {
-		case rosetta.OperationSend:
-			sendOps = append(sendOps, op)
-		case rosetta.OperationDelegate:
-			delOps = append(delOps, op)
-		}
+		operationsByType[op.Type] = append(operationsByType[op.Type], op)
 	}
-	if len(sendOps) == 2 {
-		sendMsg, err := RosettaOperationsToSdkBankMsgSend(sendOps)
-		if err != nil {
-			return nil, "", err
+
+	for opName, operations := range operationsByType {
+		if opName == rosetta.OperationFee {
+			continue
 		}
-		msgs = append(msgs, sendMsg)
-		signAddr = sendMsg.FromAddress
-	}
 
-	if len(delOps) == 2 {
-		delMsg, err := RosettaOperationsToSdkStakingMsgDelegate(delOps)
+		_, err := ir.Resolve("/" + opName)
 		if err != nil {
 			return nil, "", err
 		}
-		msgs = append(msgs, delMsg)
-		signAddr = delMsg.DelegatorAddress
+
+		if len(operations) == 2 {
+			if opName == "cosmos.bank.v1beta1.MsgSend" {
```

Comment:
hmm I think we should provide a wrapper interface for the rosetta Operations that provides a `ToMsg()` function that allows to transform it into an `sdk.Msg`. Then these conditions wouldn't be necessary

Reply:
Thanks Fede, this is still a work in progress, this is temporary.